pkg/filesystem/subvolume.go

```diff
@@ -392,6 +406,60 @@ func getOmapKey(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNam
 	return omapkey
 }
 
+// getNfsClusterName returns the cluster name from the omap.
+// csi.nfs.cluster
+// value (26 bytes) :
+// 00000000  6f 63 73 2d 73 74 6f 72  61 67 65 63 6c 75 73 74  |ocs-storageclust|
+// 00000010  65 72 2d 63 65 70 68 6e  66 73                    |er-cephnfs|
+// 0000001a
+func getNfsClusterName(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, subVol, fs string) string {
+
+	poolName, err := getMetadataPoolName(ctx, clientsets, OperatorNamespace, CephClusterNamespace, fs)
+	if err != nil || poolName == "" {
+		logging.Fatal(fmt.Errorf("pool name not found: %q", err))
+	}
+	omapval := getOmapVal(subVol)
+
+	args := []string{"getomapval", omapval, "csi.nfs.cluster", "-p", poolName, "--namespace", "csi", "/dev/stdout"}
+	cmd := "rados"
+	nfscluster, err := runCommand(ctx, clientsets, OperatorNamespace, CephClusterNamespace, cmd, args)
+	if err != nil || nfscluster == "" {
+		logging.Info("No PV found for subvolume %s: %s", subVol, err)
+		return ""
+	}
+
+	return nfscluster
+}
+
+func getNfsExportPath(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, clusterName string) string {
+
+	args := []string{"nfs", "export", "ls", clusterName}
+	cmd := "ceph"
+	nfscluster, err := runCommand(ctx, clientsets, OperatorNamespace, CephClusterNamespace, cmd, args)
+	if err != nil || nfscluster == "" {
+		logging.Info("No path found for cluster %s: %s", clusterName, err)
+		return ""
+	}
+
+	var unmarshalcluster []string
+	var nfsCluster string
```

Comment:
these variables are incorrecly named too, it is about exports, not clusters in this function